### PR TITLE
Adding certificate preview to the rewards modal and skillmap tests

### PR DESCRIPTION
**Pull request by riknoll**
More design changes from @soniakandah

### CODE REVIEW

**Review by jwunderl — APPROVED** [review]

File: skillmap/src/lib/skillMapParser.ts [hunk 1]
```diff
@@ -317,22 +366,43 @@ function inflateMapReward(section: MarkdownSection, base: Partial<MapRewardNode>
...
-
-            switch (kind) {
-                case "certificate":
-                    parsedRewards.push({
+        for (const reward of rewards.items) {
+            if (typeof reward === "string") {
+                let [kind, ...value] = reward.split(":");
+
+                switch (kind) {
+                    case "certificate":
```
Comment (shakao) [inline skillmap/src/lib/skillMapParser.ts:374]: do we need to support this case? the only map that could've been using it (before this current pr changed the syntax) was our own hour of code, right?

**Review by shakao — APPROVED** [review]

File: skillmap/src/lib/skillMapParser.ts (hunk 1 above)
Comment (riknoll) [reply skillmap/src/lib/skillMapParser.ts:374]: there are a few right now, i can remove this once release shenanigans are settled but best to keep it for now

Comment (shakao) [reply skillmap/src/lib/skillMapParser.ts:374]: sounds good! lgtm otherwise

**Review by livcheerful — APPROVED** [review]

**Comment by riknoll** [comment]
I've updated this PR to also include the necessary change for https://github.com/microsoft/pxt-arcade/issues/4320

@shakao @livcheerful please take a look at the most recent commit

**Review by dmonroym — APPROVED** [review]

**Pull request merged**